### ISSUE TITLE
MAPL_Config bug with GNU and GOCART2G

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/MAPL#951]
I was trying to run GOCART2G with GNU and I found a bug in MAPL. To wit, when you run you get:
```
At line 659 of file /discover/swdev/mathomp4/Models/GEOSgcm-feature-geos-v10.17.6-gocart2g-v2.0.0-GNU/GEOSgcm/src/Shared/@MAPL/base/MAPL_Config.F90
Fortran runtime error: End of record
```
which points to `MAPL_ConfigSetAttribute_reals32` a function added by @atrayano:
https://github.com/GEOS-ESM/MAPL/blob/d5009302d5ebac669cc7ff93db5134154a2c7a88/base/MAPL_Config.F90#L649-L664

The issue seems to be with the 'magic number' 15. If you make a small little Fortran program:
```fortran
program main

   implicit none

   character(len=15) :: tmpStr
   real :: value(5)
   integer :: i

   value = [500, 600, 700, 800, 900]

   do i = 1, size(value)
      write(*,*) "i: ", i, "value: ", value(i)
      write(tmpStr,*) value(i)
      write(*,*) 'tmpStr:', tmpStr
   end do

end program main
```
This works with Intel:
```console
❯ ifort test.F90 && ./a.out
 i:            1 value:    500.0000
 tmpStr:   500.0000
 i:            2 value:    600.0000
 tmpStr:   600.0000
 i:            3 value:    700.0000
 tmpStr:   700.0000
 i:            4 value:    800.0000
 tmpStr:   800.0000
 i:            5 value:    900.0000
 tmpStr:   900.0000
```
but fails with GNU:
```console
❯ gfortran test.F90 && ./a.out
 i:            1 value:    500.000000
At line 26 of file test.F90
Fortran runtime error: End of record
```

So there are a couple of things. One, it looks like Intel is doing `F11.4` so we could change:
https://github.com/GEOS-ESM/MAPL/blob/d5009302d5ebac669cc7ff93db5134154a2c7a88/base/MAPL_Config.F90#L659
to:
```fortran
        write(tmpStr, '(F11.4)') value(i) ! ALT: check if enough space to write
```

If so, we'd also probably want to look at the magic number in the int-vector case:
https://github.com/GEOS-ESM/MAPL/blob/d5009302d5ebac669cc7ff93db5134154a2c7a88/base/MAPL_Config.F90#L616
and maybe do:
```fortran
        write(tmpStr, '(I0)') value(i) ! ALT: check if enough space to write
```
??

Perhaps also, we should have unit tests for this in `base/tests/Test_MAPL_Config.pf` which only has a test for string:
https://github.com/GEOS-ESM/MAPL/blob/d5009302d5ebac669cc7ff93db5134154a2c7a88/base/tests/Test_MAPL_Config.pf#L8-L9

Not sure. @tclune might wish to weigh in.

**Comment by mathomp4** [comment]
Note, I did some testing with Atanas and it turns out the magic numbers are:

- Intel: 15
- NAG: 16
- NVHPC: 16
- GNU: 18

18 is ugly, but GNU is weird?

**Comment by mathomp4** [comment]
@atrayano Are you okay with this being 18 characters as that's the GNU magic number?

**Comment by tclune** [comment]
If not, I can work with you to convert this to a deferred length string where we explicitly allocate to a length long enough to contain whatever needs to go in.

**Comment by atrayano** [comment]
Fixed in #1164
